src/context/web3email.tsx

```diff
@@ -0,0 +1,273 @@
+    chainId: parseInt(process.env.NEXT_PUBLIC_NETWORK_ID as string),
+  });
+
+  async function handleAgreeAndClose() {
+    if (user && provider && signer) {
+      try {
+        const cid = await postToIPFS(
+          JSON.stringify({
+            title: user.description?.title,
+            role: user.description?.role,
+            image_url: user.description?.image_url,
+            video_url: user.description?.video_url,
+            name: user.description?.name,
+            about: user.description?.about,
+            skills: user.description?.skills_raw,
+            mail: {
+              proposalConsent: proposalConsent,
+            },
+          }),
+        );
+
+        console.log('cid', cid);
+
+        let tx;
+        if (isActiveDelegate) {
+          const response = await delegateUpdateProfileData(user.id, user.address, cid);
+          tx = response.data.transaction;
+        } else {
+          const contract = new ethers.Contract(
+            config.contracts.talentLayerId,
+            TalentLayerID.abi,
+            signer,
+          );
+          tx = await contract.updateProfileData(user.id, cid);
+        }
+
+        await createMultiStepsTransactionToast(
+          {
+            pending: 'Updating profile...',
+            success: 'Congrats! Your preferences has been updated',
+            error: 'An error occurred while updating your preferences',
+          },
+          provider,
+          tx,
+          'user',
+          cid,
+        );
+        setShow(false);
+        setIsRedirect(true);
+      } catch (error) {
+        showErrorTransactionToast(error);
+      }
+    } else {
+      openConnectModal();
+    }
+  }
+
+  async function fetchProtectedMail() {
+    const fetchProtectedDataArg: FetchProtectedDataParams = {
+      owner: user?.address,
+    };
+    console.log('fetchProtectedDataArg', fetchProtectedDataArg);
+
+    // if user?.address is undefined it will fetch the whole protected data
+    if (user?.address) {
+      const tx = await fetchProtectedData(fetchProtectedDataArg);
+      console.log('fetchProtectedDataTest:', tx.data.data.fetchProtectedData);
+      setProtectedMails(tx.data.data.fetchProtectedData);
+    }
+  }
+
+  useEffect(() => {
+    fetchProtectedMail();
+  }, [user]);
+
+  useEffect(() => {
+    if (protectedMails.length > 0) {
+      setIsRedirect(true);
+    }
+  }, [protectedMails]);
+
+  interface IFormValues {
+    email: string;
+  }
+
+  const initialValues = {
+    email: '',
+  };
+
+  const validationSchema = Yup.object({
+    email: Yup.string().email('Invalid email address').required('Required'),
+  });
+
+  const onSubmit = async (
+    values: IFormValues,
+    {
+      setSubmitting,
+      resetForm,
+    }: {
+      setSubmitting: (isSubmitting: boolean) => void;
+      resetForm: () => void;
+    },
+  ) => {
+    setIsLoading(true);
+    try {
+      const protectedData = await dataProtector({ data: { email: values.email } });
```

Comment:
- Email must never be passed to the backend, that the opposite of the philosophy of web3 email 
- In this API, it's now the app that protect the data for the user, that's also not the logic of web3 email 

**Todo :** 
- Remove this API 
- Protect the data from the frontend